src/System.Reflection.Emit/tests/FieldBuilder/FieldBuilderSetConstant.cs

```diff
@@ -0,0 +1,353 @@
+            VerificationHelper(field, null, typeof(ArgumentException));
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_UInt32", typeof(UInt32), FieldAttributes.Public);
+            VerificationHelper(field, null, typeof(ArgumentException));
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_UInt64", typeof(UInt64), FieldAttributes.Public);
+            VerificationHelper(field, null, typeof(ArgumentException));
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_Single", typeof(Single), FieldAttributes.Public);
+            VerificationHelper(field, null, typeof(ArgumentException));
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_Double", typeof(Double), FieldAttributes.Public);
+            VerificationHelper(field, null, typeof(ArgumentException));
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_DateTime", typeof(DateTime), FieldAttributes.Public);
+            VerificationHelper(field, null, typeof(ArgumentException));
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_Char", typeof(Char), FieldAttributes.Public);
+            VerificationHelper(field, null, typeof(ArgumentException));
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_String", typeof(string), FieldAttributes.Public);
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_FBTestEnum", typeof(FBTestEnum), FieldAttributes.Public);
+            VerificationHelper(field, null, typeof(ArgumentException));
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_Decimal", typeof(Decimal), FieldAttributes.Public);
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_Object", typeof(Object), FieldAttributes.Public);
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+        }
+
+        [Fact]
+        public void NegTest3()
+        {
+            try
+            {
+                FieldBuilder field = TypeBuilder.DefineField("Field_NegTest3", typeof(FBTestStruct), FieldAttributes.Public);
+                VerificationHelper(field, null, typeof(ArgumentException));
+            }
+            catch (InvalidOperationException)
```

Comment:
In internal test infrastructure, we were using a static typebuilder, hence if createtype had already been called on some other method, this would have thrown invalidoperationexception. Doesn't matter now in current design.